lib/puppet/transaction/persistence.rb

```diff
@@ -0,0 +1,81 @@
+require 'yaml'
+require 'puppet/util/yaml'
+
+# A persistence store implementation for storing information between
+# transaction runs for the purposes of information inference (such
+# as calculating corrective_change).
+# @api private
+class Puppet::Transaction::Persistence
```

Comment:
We went down that road also, kylo and eric0 had concerns about extending the report format this way, when we had no intention of using it directly (ie. PuppetDB wasn't going to consume these system values, nor was it ready for that). Since its a rather public API, we decided it best to avoid that kind of thinking for later.